From f3f8cf415de8374102dd1f88276f124a0ff46800 Mon Sep 17 00:00:00 2001
From: Andrew Lunn <andrew@lunn.ch>
Date: Mon, 5 May 2025 18:01:02 -0500
Subject: [PATCH 10/30] net: phy: maxio: Use phy_read_paged()/phy_write_paged()

Use the core helpers to do paged access. The page is then kept at 0 by
default. Remove the local helpers which are now redundant.

Signed-off-by: Andrew Lunn <andrew@lunn.ch>
---
 drivers/net/phy/maxio.c | 63 +++++++----------------------------------
 1 file changed, 11 insertions(+), 52 deletions(-)

diff --git a/drivers/net/phy/maxio.c b/drivers/net/phy/maxio.c
index 8f2c5c8a4..d5a597c5d 100755
--- a/drivers/net/phy/maxio.c
+++ b/drivers/net/phy/maxio.c
@@ -35,71 +35,35 @@ static int maxio_write_page(struct phy_device *phydev, int page)
 	return __phy_write(phydev, MAXIO_PAGE_SELECT, page);
 }
 
-static int maxio_read_paged(struct phy_device *phydev, int page, u32 regnum)
-{
-	int ret = 0, oldpage;
-
-	oldpage = phy_read(phydev, MAXIO_PAGE_SELECT);
-	if (oldpage >= 0) {
-		phy_write(phydev, MAXIO_PAGE_SELECT, page);
-		ret = phy_read(phydev, regnum);
-    }
-	phy_write(phydev, MAXIO_PAGE_SELECT, oldpage);
-
-	return ret;
-}
-
-static int maxio_write_paged(struct phy_device *phydev, int page, u32 regnum, u16 val)
-{
-	int ret = 0, oldpage;
-
-	oldpage = phy_read(phydev, MAXIO_PAGE_SELECT);
-	if (oldpage >= 0) {
-		phy_write(phydev, MAXIO_PAGE_SELECT, page);
-		ret = phy_write(phydev, regnum, val);
-	}
-	phy_write(phydev, MAXIO_PAGE_SELECT, oldpage);
-
-	return ret;
-}
-
 static int maxio_mae0621a_clk_init(struct phy_device *phydev)
 {
-	u32 workmode,clkmode,oldpage;
+	u32 workmode,clkmode;
 	int ret;
 
-	oldpage = phy_read(phydev, MAXIO_PAGE_SELECT);
-	if (oldpage == 0xFFFF)	{
-		oldpage = phy_read(phydev, MAXIO_PAGE_SELECT);
-	}
-
 	ret = genphy_soft_reset(phydev);
 	if (ret < 0)
 		return ret;
 
 	//get workmode
-	phy_write(phydev, MAXIO_PAGE_SELECT, 0xa43);
-	workmode = phy_read(phydev, MAXIO_MAE0621A_WORK_STATUS_REG);
+	workmode = phy_read_paged(phydev, 0xa43,
+				  MAXIO_MAE0621A_WORK_STATUS_REG);
 
 	//get clkmode
-	phy_write( phydev,  MAXIO_PAGE_SELECT, 0xd92 );
-	clkmode = phy_read( phydev, MAXIO_MAE0621A_CLK_MODE_REG );
+	clkmode = phy_read_paged(phydev, 0xd92, MAXIO_MAE0621A_CLK_MODE_REG);
 
 	//abnormal
 	if (0 == (workmode&BIT(5))) {
 		if (0 == (clkmode&BIT(8))) {
 			//oscillator
-			phy_write(phydev, 0x02, clkmode | BIT(8));
+			phy_write_paged(phydev, 0xd92, 0x02, clkmode | BIT(8));
 			printk("****maxio_mae0621a_clk_init**clkmode**0x210a: 0x%x\n", phydev->phy_id);
 		} else {
 			//crystal
 			printk("****maxio_mae0621a_clk_init**clkmode**0x200a: 0x%x\n", phydev->phy_id);
-			phy_write(phydev, 0x02, clkmode &(~ BIT(8)));
+			phy_write_paged(phydev, 0xd92, 0x02,
+					clkmode &(~ BIT(8)));
 		}
 	}
-	phy_write(phydev, MAXIO_PAGE_SELECT, 0x0);
-
-	phy_write(phydev, MAXIO_PAGE_SELECT, oldpage);
 
 	return 0;
 }
@@ -115,7 +79,7 @@ static int maxio_mae0621a_config_init(struct phy_device *phydev)
 	phy_disable_eee(phydev);
 
 	//enable auto_speed_down
-	ret = maxio_write_paged(phydev, 0xd8f, 0x0, 0x300 );
+	ret = phy_write_paged(phydev, 0xd8f, 0x0, 0x300 );
 
 	//adjust TX/RX delay
 	switch (phydev->interface) {
@@ -135,13 +99,13 @@ static int maxio_mae0621a_config_init(struct phy_device *phydev)
 		goto delay_skip;
 	}
 
-	ret = maxio_read_paged(phydev, 0xd96, 0x0);
+	ret = phy_read_paged(phydev, 0xd96, 0x0);
 	if (ret < 0) {
 		dev_err(dev, "Failed to update the TX delay register\n");
 		return ret;
 	}
 
-	ret =maxio_write_paged(phydev, 0xd96, 0x0, val|ret );
+	ret = phy_write_paged(phydev, 0xd96, 0x0, val|ret );
 	if (ret < 0) {
 		dev_err(dev, "Failed to update the TX delay register\n");
 		return ret;
@@ -156,8 +120,6 @@ static int maxio_mae0621a_config_init(struct phy_device *phydev)
 	if (ret < 0)
 		return ret;
 
-	phy_write(phydev, MAXIO_PAGE_SELECT, 0x0);
-
 	return 0;
 }
 
@@ -173,10 +135,7 @@ static int maxio_mae0621a_resume(struct phy_device *phydev)
 
 static int maxio_mae0621a_suspend(struct phy_device *phydev)
 {
-	genphy_suspend(phydev);
-	phy_write(phydev, MAXIO_PAGE_SELECT ,0);
-
-	return 0;
+	return genphy_suspend(phydev);
 }
 
 static int maxio_mae0621a_probe(struct phy_device *phydev)
-- 
2.39.5

